nx-X11/programs/Xserver/composite/compinit.c

```diff
@@ -139,192 +218,190 @@ compFindVisuallessDepth (ScreenPtr pScreen, int d)
     return 0;
 }
 
+/*
+ * Add a list of visual IDs to the list of visuals to implicitly redirect.
+ */
+static Bool
+compRegisterAlternateVisuals(CompScreenPtr cs, VisualID * vids, int nVisuals)
+{
+    VisualID *p;
+
+#ifndef NXAGENT_SERVER
```

Comment:
Is this ifdef NX specific? Or actually related to another feature lacking in the Xserver code tree? I'd say let's only use NXAGENT_SERVER for changes to the Xserver code that have to do with better performance on remote connection and such stuff.

If we have to work around missing Xserver features, let's name that explicitly. Let's use one or more definitions where each definition stands for a lacking Xserver feature.

Whenever we get that feature in, we may simply undef that definition.